### PR TITLE
feat: upgrade etherjs v6

### PR DESCRIPTION
**Pull request by cheng521521**
feat:upgrade etherjs v6 - 13-decentralized-exchange

### CODE REVIEW

**Review by yingjingyang — APPROVED** [review]

File: basic/13-decentralized-exchange/uniswap-v1-like/hardhat.config.js [hunk 1]
```diff
@@ -1,7 +1,5 @@
-require("@nomiclabs/hardhat-waffle");
-const { utils } = require("ethers");
-const { GAS_PRICE } = require("./utils")
-
+require("@nomicfoundation/hardhat-toolbox");
+var ethers = require('ethers').ethers;
```
Comment (stevending1st) [inline basic/13-decentralized-exchange/uniswap-v1-like/hardhat.config.js:2]: I'm not sure it's better to write this way:

```suggestion
const { ethers } = require('ethers');
```

File: basic/13-decentralized-exchange/uniswap-v1-like/test/Exchange.test.js [hunk 2]
```diff
@@ -1,24 +1,30 @@
...
-  const exchangeAddress = await factory.connect(sender).callStatic.createExchange(tokenAddress);
 
-  await factory.connect(sender).createExchange(tokenAddress);
+  console.log(tokenAddress, sender.address);
+  const exchangeAddress = await factory.connect(sender).createExchange(tokenAddress);
+  exchangeAddress.wait();
 
+  const address = await factory.connect(sender).getExchange(tokenAddress);
+  console.log("token address-" + tokenAddress + "-exchange address-" + address);
+  console.log(address);
```
Comment (stevending1st) [inline basic/13-decentralized-exchange/uniswap-v1-like/test/Exchange.test.js:24]: delete the log ?

File: basic/13-decentralized-exchange/uniswap-v1-like/utils/index.js [hunk 3]
```diff
@@ -1,17 +1,19 @@
-const { ethers } = require("ethers");
-const { BigNumber } = ethers;
+const hre = require('hardhat');
+const ethers = hre.ethers;
 
-const GAS_PRICE = BigNumber.from("1000000000");
+const GAS_PRICE = 1000000000n;
 
 const getGasFeeFromTx = async (tx) => {
+  console.log(33333);
```
Comment (stevending1st) [inline basic/13-decentralized-exchange/uniswap-v1-like/utils/index.js:7]: delete the log?

File: basic/13-decentralized-exchange/uniswap-v1-like/utils/index.js [hunk 4]
```diff
@@ -1,17 +1,19 @@
...
   if (tx && typeof tx.wait === 'function') {
     const { gasUsed } = await tx.wait()
-    return gasUsed.mul(GAS_PRICE)
+    console.log(gasUsed);
```
Comment (stevending1st) [inline basic/13-decentralized-exchange/uniswap-v1-like/utils/index.js:10]: delete the log?

File: basic/13-decentralized-exchange/uniswap-v1-like/test/Exchange.test.js (hunk 2 above)
Comment (cheng521521) [reply basic/13-decentralized-exchange/uniswap-v1-like/test/Exchange.test.js:24]: I won't delete the log statement console.log("token address-" + tokenAddress + "-exchange address-" + address); during my study. This log helps me debug the program and understand its principles.

File: basic/13-decentralized-exchange/uniswap-v1-like/utils/index.js (hunk 3 above)
Comment (cheng521521) [reply basic/13-decentralized-exchange/uniswap-v1-like/utils/index.js:7]: yes,you are correct

File: basic/13-decentralized-exchange/uniswap-v1-like/utils/index.js (hunk 4 above)
Comment (cheng521521) [reply basic/13-decentralized-exchange/uniswap-v1-like/utils/index.js:10]: yes,you are correct

File: basic/13-decentralized-exchange/uniswap-v1-like/hardhat.config.js (hunk 1 above)
Comment (cheng521521) [reply basic/13-decentralized-exchange/uniswap-v1-like/hardhat.config.js]: yes,you are correct.thx

**Review by 0x-stan — APPROVED** [review]

**Review by yingjingyang — APPROVED** [review]

**Pull request merged**